tests/test_api_routes.py

```diff
@@ -0,0 +1,155 @@
+))
+def test_search_filter_status(client, mock_retriever_model, status, expected_statuses):
+    status_string = '&'.join(f'status={s}' for s in status)
+    response = client.get('/api/search?top=10&' + status_string)
+    assert response.status_code == 200
+    assert {i['status'] for i in response.json} == expected_statuses
+
+
+@pytest.mark.parametrize('showLive', ('true', 'TrUe', '1'))
+def test_search_showLive_selected(client, mock_retriever_model, showLive):
+    response = client.get(f'/api/search?top=10&showLive={showLive}')
+    assert response.status_code == 200
+    assert {i['status'] for i in response.json} == {'Live on site'}
+
+
+@pytest.mark.parametrize('showLive', ('0', 'no', 'false', 'bla', 'asdadsa'))
+def test_search_showLive_disabled(client, mock_retriever_model, showLive):
+    response = client.get(f'/api/search?top=10&showLive={showLive}')
+    assert response.status_code == 200
+    assert {i['status'] for i in response.json} == {'In progress', 'Not started'}
+
+
+def test_search_status_overrides_showLive(client, mock_retriever_model):
+    response = client.get(f'/api/search?top=10&showLive=true&status=In review')
+    assert response.status_code == 200
+    assert {i['status'] for i in response.json} == {'In review'}
+
+
+SEARCH_ENDPOINT = 'https://nlp.stampy.ai/api/search'
+
+@pytest.mark.live
+def test_search_known_question():
+    response = requests.get(SEARCH_ENDPOINT, params={'query': 'What is AI safety?', 'showLive': 1})
+    questions = response.json()
+    assert questions
+    assert 'What is AI safety?' in [i['title'] for i in questions]
+    assert {i['status'] for i in questions} == {'Live on site'}
+
+
+@pytest.mark.live
+def test_search_known_question_live():
+    response = requests.get(SEARCH_ENDPOINT, params={'query': 'What is AI safety?', 'showLive': 0})
+    questions = response.json()
+    assert questions
+    assert 'Live on site' not in {i['status'] for i in questions}
+
+
+@pytest.mark.live
+def test_search_all():
+    response = requests.get(SEARCH_ENDPOINT, params={'query': 'Something random', 'top': 20, 'status': 'all'})
+    questions = response.json()
+    assert len(questions) == 20
+
+    statuses = {i['status'] for i in questions}
+    # This is a potential cause of Heisenbugs, seeing as it's possible that the top 20
+    # questions returned are all live. But this isn't all that likely, so should be fine...?
+    assert len(statuses) > 1
+    assert 'Live on site' in statuses
+
+
+@pytest.mark.live
+def test_search_specific_status():
+    response = requests.get(
+        SEARCH_ENDPOINT,
+        params={'query': 'Something random', 'top': 20, 'status': ['Not started', 'In progress']}
+    )
+    questions = response.json()
+    assert len(questions) == 20
+
+    # This is a potential cause of Heisenbugs, seeing as it's possible that the top 20
+    # questions returned are all live. But this isn't all that likely, so should be fine...?
```

Comment:
Yes, good note to put there. Setting the number of returned entries to 20 perhaps buys us a bit of time but eventually this may fail as the set of questions in the FAQ grows.